src/providers/dart_commands.ts

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+import * as as from "../analysis/analysis_server_types";
+import * as editors from "../editors";
+import * as vs from "vscode";
+import { Analyzer } from "../analysis/analyzer";
+
+export class DartCommands  {
+	private analyzer: Analyzer;
+	constructor(context: vs.ExtensionContext, analyzer: Analyzer) {
+		this.analyzer = analyzer;
+
+        context.subscriptions.push(
+            vs.commands.registerTextEditorCommand("dart.organizeDirectives", this.organizeDirectives, this)
+        );
+	}
+
+    organizeDirectives(editor: vs.TextEditor, editBuilder: vs.TextEditorEdit) {
+        if (!editors.hasActiveDartEditor()) {
+            vs.window.showWarningMessage("No active Dart editor.");
+            return;
+        }
+
+        this.analyzer.editOrganizeDirectives({ file: editor.document.fileName }).then((response) => {
+            let edit: as.SourceFileEdit = response.edit;
+			if (edit.edits.length == 0) {
+                vs.window.showInformationMessage("No changes from organize directives.");
```

Comment:
I think maybe it'd be best to just be silent if there are no changes to make. We wouldn't alert the user if a Format Document operation didn't make any changes, so I don't think this is much different (IDEs I've used don't alert here, but I haven't used many!).

Thoughts?